When using robot_radius, footprint is incorrectly set by costmap2d_ros

When using robot_radius, the footprint gets set by move_base to [] if the footprint parameter is not set. This means that the second time you launch the navigation stack, the footprint will be improper and an error appears:

```
[ERROR] [1402335344.906978220, 66.013000000]: You must specify at least three points for the robot footprint, reverting to previous footprint.
```

My current work around has been to add this to the launch file:

```
<rosparam command="delete" ns="move_base" />
```

--- a/costmap_2d/src/footprint.cpp
+++ b/costmap_2d/src/footprint.cpp
@@ -225,15 +225,18 @@ std::vector<geometry_msgs::Point> makeFootprintFromParams(ros::NodeHandle& nh)
       if (makeFootprintFromString(std::string(footprint_xmlrpc), points))
       {
         writeFootprintToParam(nh, points);
+        return points;
       }
     }
     else if (footprint_xmlrpc.getType() == XmlRpc::XmlRpcValue::TypeArray)
     {
       points = makeFootprintFromXMLRPC(footprint_xmlrpc, full_param_name);
       writeFootprintToParam(nh, points);
+      return points;
     }
   }
-  else if (nh.searchParam("robot_radius", full_radius_param_name))
+
+  if (nh.searchParam("robot_radius", full_radius_param_name))
   {
     double robot_radius;
     nh.param(full_radius_param_name, robot_radius, 1.234);

--- a/costmap_2d/src/footprint.cpp
+++ b/costmap_2d/src/footprint.cpp
@@ -219,7 +219,8 @@ std::vector<geometry_msgs::Point> makeFootprintFromParams(ros::NodeHandle& nh)
   {
     XmlRpc::XmlRpcValue footprint_xmlrpc;
     nh.getParam(full_param_name, footprint_xmlrpc);
-    if (footprint_xmlrpc.getType() == XmlRpc::XmlRpcValue::TypeString)
+    if (footprint_xmlrpc.getType() == XmlRpc::XmlRpcValue::TypeString &&
+        footprint_xmlrpc != "" && footprint_xmlrpc != "[]")
     {
       if (makeFootprintFromString(std::string(footprint_xmlrpc), points))
       {

